packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,268 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const error = require('../error');
+const isA = require('joi');
+const ScopeSet = require('fxa-shared').oauth.scopes;
+
+module.exports = (log, db, config, customs, oauthdb, subhub) => {
+  const totpUtils = require('./utils/totp')(log, config, db);
+
+  const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
+    'https://identity.mozilla.com/apps/subscriptions';
+
+  async function handleAuth(auth, fetchAccount = true) {
+    let scope, uid, email;
+
+    if (auth.strategy === 'sessionToken') {
+      // sessionToken strategy
+      scope = { contains: () => true };
+      ({ uid, email} = auth.credentials);
+      const hasEnabledToken = await totpUtils.hasTotpToken(uid);
+      if (
+        hasEnabledToken 
+        && (
+          sessionToken.tokenVerificationId
+          || sessionToken.authenticatorAssuranceLevel <= 1
+        )
+      ) {
+        throw errors.unverifiedSession();
+      }
+    } else {
+      // oauthToken strategy
+      scope = ScopeSet.fromArray(auth.credentials.scope);
+      if (! scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
+        throw error.invalidToken('Invalid authentication scope in token');
+      }
+      uid = auth.credentials.user;
+      if (fetchAccount) {
+        const account = await db.account(uid);
+        email = account.primaryEmail.email;
+      }
+    }
+
+    return { scope, uid, email };
+  }
+
+  return [
+    {
+      method: 'POST',
+      path: '/subscriptions/token',
```

Comment:
This is how & where I've made an attempt at issuing the OAuth token for use by the off-domain subscription management pages. I copied it in part from the `/oauth/token` but with hardcoded values. Could use some scruitiny here to see if this could be done better.

FWIW, I'm imagining it'll be called from 1) FxA subscription settings page when the "manage" button is clicked for subscriptions and 2) on an FxA page linked-to from external products that will redirect to a subscription signup page.

Reply:
Yeah, that would simplify things here. I thought maybe having a special-purpose route just for issuing this particular token would be a good idea, but it simplifies things not to have it.